web/static/js/components/simulation/MessagesList.jsx

```diff
@@ -50,33 +39,42 @@ export class MessagesList extends Component<MessagesListProps> {
     this.scrollToBottom()
   }
 
-  componentDidUpdate() {
-    this.scrollToBottom()
+  componentDidUpdate(prevProps) {
+    // only autoscroll to bottom if the list of messages changed (we assume the
+    // list can only grow or shrink, not changed in place):
+    if (prevProps.messages.length !== this.props.messages.length) {
+      this.scrollToBottom()
+    }
+  }
+
+  mustTruncate() {
+    return this.props.truncateAt && this.props.truncateAt > 0
   }
 
   render() {
-    const groupBy = (elems, func) => {
-      const lastElem = (collection: Array<any>) => collection[collection.length - 1]
-
-      return elems.reduce(function (groups, elem) {
-        const lastGroup = lastElem(groups)
-        if (groups.length == 0 || func(lastElem(lastGroup)) != func(elem)) {
-          groups.push([elem])
-        } else {
-          lastGroup.push(elem)
-        }
-        return groups
-      }, [])
-    }
+    let { messages } = this.props
 
-    const { messages } = this.props
-    const groupedMessages = groupBy(messages, (message: ChatMessage) => message.type)
+    if (this.mustTruncate()) {
+      messages = this.truncate(messages)
+    }
 
     return (
       <div className="chat-window-body">
         <ul>
-          {groupedMessages.map((messages, ix) => (
-            <MessageBulk key={`msg-bulk-${ix}`} messages={messages} />
+          {messages.map((message, index) => (
```

Comment:
I'm dropping the `MessageBulk` component because it didn't add much, and actually required a bunch of useless indirection. 

BUT now that I'm writing this, maybe I should have have keep it, and put the individual `truncated` state there, instead of a using a `truncated` array in MessagesList. Hum :thinking: yeah, that would be better :facepalm: 

I also dropped grouping the subsequent AO and AT messages, since it led to invalid DOM (UL > DIV > LI) and I failed to notice any visual difference?